src/agent0/chainsync/dashboard/build_ticker.py

```diff
@@ -2,34 +2,116 @@
+    trade_events = trade_events.merge(block_to_timestamp, how="left", on="block_number")
+
+    rename_dict = {
+        "timestamp": "Timestamp",
+        "block_number": "Block Number",
+        "username": "User",
+        "wallet_address": "Wallet",
+        "event_type": "Trade",
+        "token_id": "Token",
+        "token_delta": "Token Change",
+        "base_delta": "Base Change",
+        "as_base": "As Base",
+    }
+    trade_events = trade_events[list(rename_dict.keys())].rename(columns=rename_dict)
 
-    ticker_data = ticker_data.copy()
-    ticker_data = ticker_data.drop("id", axis=1)
-    ticker_data.insert(2, "username", mapped_addrs["username"])
-    ticker_data.columns = ["Block Number", "Timestamp", "User", "Wallet", "Method", "Token Deltas"]
     # Shorten wallet address string
-    ticker_data["Wallet"] = mapped_addrs["abbr_address"]
-    # Drop rows with nonexistant wallets
-    ticker_data = ticker_data.dropna(axis=0, subset="Wallet")
-    return ticker_data
+    trade_events["Wallet"] = mapped_addrs["abbr_address"]
+
+    # Sort latest first
+    return trade_events
+
+
+def build_wallet_ticker(
+    trade_events: pd.DataFrame,
+    user_map: pd.DataFrame,
+    hyperdrive_addr_map: pd.DataFrame,
+    block_to_timestamp: pd.DataFrame,
+) -> pd.DataFrame:
+    """Show recent trades wrt a wallet.
+
+    Arguments
+    ---------
+    trade_events: pd.DataFrame
+        The dataframe resulting from `get_trade_events`.
+    user_map: pd.DataFrame
+        A dataframe containing the mapping of wallet addresses to usernames.
+    hyperdrive_addr_map: pd.DataFrame
+        A dataframe containing the mapping of hyperdrive addresses to hyperdrive names.
+    block_to_timestamp: pd.DataFrame
+        A dataframe containing the mapping of block number to timestamp.
+
+    Returns
+    -------
+    pd.DataFrame
+        The filtered transaction data based on what we want to view in the ticker for a specific wallet.
+    """
+
+    mapped_addrs = map_addresses(trade_events["wallet_address"], user_map)
+    trade_events = trade_events.copy()
+    trade_events["username"] = mapped_addrs["username"]
+
+    # Do lookup from address to name
+    hyperdrive_name = (
+        trade_events["hyperdrive_address"]
+        .to_frame()
+        .merge(hyperdrive_addr_map, how="left", left_on="hyperdrive_address", right_on="hyperdrive_address")
+    )["name"]
+
+    trade_events["hyperdrive_name"] = hyperdrive_name
+
+    # Look up block to timestamp
+    trade_events = trade_events.merge(block_to_timestamp, how="left", on="block_number")
+
+    rename_dict = {
+        "timestamp": "Timestamp",
+        "block_number": "Block Number",
+        "hyperdrive_name": "Hyperdrive Name",
+        "hyperdrive_address": "Hyperdrive Address",
+        "username": "User",
+        "wallet_address": "Wallet",
+        "event_type": "Trade",
+        "token_id": "Token",
+        "token_delta": "Token Change",
+        "base_delta": "Base Change",
```

Comment:
this isn't always `base` right? Could be `shares` or `base` depending on `as_base`?

Reply:
Nice catch, resolved by removing `as_base` and adding in `vault_share_delta`